### ISSUE TITLE
shaders cleanup

### ISSUE THREAD

**Issue by rukai** [vulkano-rs/vulkano#1039]
Once https://github.com/vulkano-rs/vulkano/pull/947 is merged then, there a bunch of work I want to do on the vulkano-shaders/vulkano-shader-derive crates, in terms of updating dependencies and cleaning things up. So figured I would list my plans here:
1.  delete fn build_glsl (it was used for pre proc-macros)
2.  deprecate glsl-to-spirv (change readme + publish)
3.  upgrade to proc_macro2
4.  change panics to errors as per the syn lazy-static example
5.  deprecate vulkano-shaders (change readme + publish)
6.  Investigate if *-derive crates will be made unneccessary. Then decide to remove comments calling for its merge or merge it.
        Actually nah because vulkano-shaders is generating source code as a string this will need to be merged, potentially even when I upgrade to syn 0.15
7.  Pass file names to the input_file_name argument. Filename will be either:
    +   *.glsl
    +   *.rs (embedded)
8.  Replace vulkano-shaders/src/parse.rs with rspirv stuff

**Comment by rukai** [comment]
https://github.com/vulkano-rs/vulkano/issues/945 Just came across this issue, posting it here so I can find it again.

**Comment by TheEdward162** [comment]
I don't know if this is the right place to ask this, but is there a plan to support shaderc #include directive? According to the shaderc-rs docs, it's just a matter of supplying a loader function in [compile_options](https://github.com/vulkano-rs/vulkano/blob/c57289f53611acbfba10a6d7dd9f8b40553ae0fa/vulkano-shaders/src/codegen.rs#L30) and it would really be a big help when reusing shader code.

**Comment by rukai** [comment]
Thanks for pointing that out.
I haven't seen it before but it would be nice to have.

Basic design:
*   When the shader is passed as a string in the `shaders!` macro we dont set the callback at all.
*   When the second arg is IncludeType::Standard the callback would return Err
*   When the second arg is IncludeType::Relative it would attempt to read a shader string using the path in the first arg starting from the directory that the compiled shader is in.
